### ISSUE TITLE
Incorporate good ideas from DeMix-Q

### ISSUE THREAD

**Issue by hendrikweisser** [OpenMS/OpenMS#1952]
DeMix-Q is a label-free quantification workflow that was recently published in MCP (http://www.mcponline.org/content/15/4/1467.full; shout-out to @hroest for pointing out the paper). It is largely based on TOPP tools, but adds some good ideas - some of which we should incorporate into OpenMS. In particular:
1. Filling of missing values in a consensus map using EICExtractor. We already have SeedListGenerator, which can generate lists of RT and m/z positions of missing features when given a consensus map. We should extend EICExtractor to accept featureXML (as generated by SeedListGenerator) for its "pos" input, and write output in featureXML format (in addition to CSV). (Alternatively, EICExtractor could directly read the consensus map and write out an augmented version of it, with missing values filled in, but that would be a bigger change.)
2. RT-dependent intensity correction for features (see Fig. 3 in the paper). This could be readily added to ConsensusMapNormalizer as an alternative algorithm. @cbielow is working on non-linear RT-dependent mass correction (in InternalCalibration?); I assume that code could be easily repurposed for intensity correction.

**Comment by hroest** [comment]
Hi Hendrik,

1. basically the described workflow uses some featurelinker (e.g. FeatureLinkerQT) and then uses the generated consensusXML to generate coordiantes for the EICExtractor - so maybe having consensusXML as input to EICExtractor may be even more useful? but still you would have to do some heavy processing to convert the extracted EIC into something like a feature (e.g. detection of start/end) so its not as easy. 

2. we have been doing this for SWATH MS for a while with good results. Basically, we fit a lowess model to the RT - Intensity space and then correct on that. I am not sure whether the approach in the paper is better but it seems to rely a lot on which other runs are present. Our approach is per-run and thus more independent from which other runs are present (I dont quite like the idea of depending so much on which other runs are there)

**Comment by hendrikweisser** [comment]
> but still you would have to do some heavy processing to convert the extracted EIC into something like a feature (e.g. detection of start/end) so its not as easy

I don't think so. I would just turn the whole extracted region into a feature and not worry about detecting the ends.

> Basically, we fit a lowess model to the RT - Intensity space and then correct on that.

Do you do that on the feature level or on the raw data?

I still think their approach would be worth implementing, especially since we already have the ConsensusMapNormalizer tool. Then we could compare it to your approach. (Is it implemented in OpenMS?)

**Comment by timosachsenberg** [comment]
@hroest can you point out the code for 2. Do you think this would be applicable in ConsensusMap normalizer?
